synapse/storage/databases/main/stats.py

```diff
@@ -554,7 +554,17 @@ def _fetch_current_state_stats(
             "get_initial_state_for_room", _fetch_current_state_stats
         )
 
-        state_event_map = await self.get_events(event_ids, get_prev_content=False)  # type: ignore[attr-defined]
+        try:
+            state_event_map = await self.get_events(event_ids, get_prev_content=False)  # type: ignore[attr-defined]
+        except Exception as e:
```

Comment:
This being `Exception` seems kind of meh if there are logic errors in that function, but that's what we raise.

Reply:
Is it worth quickly adding a new exception class so we can specifically catch it?

Reply:
I think so, I'll do that. 💯

Reply:
ddeddae77a775714dfbea2edd96218fc2a49692e